### PR TITLE
Use new methods introduced in keys.go

### PR DESCRIPTION
**Pull request by priteshbandi**
This PR depends on https://github.com/notaryproject/notation-go/pull/246 so please review https://github.com/notaryproject/notation-go/pull/246 first.

### CODE REVIEW

File: cmd/notation/key.go [hunk 1]
```diff
@@ -294,26 +225,19 @@ func deleteKeys(ctx context.Context, opts *keyDeleteOpts) error {
...
-		if prevDefault == name {
-			signingKeys.Default = ""
+	var deletedNames []string 
+	var prevDefault string
+	exec := func(s *config.SigningKeys) error {
+		prevDefault = *s.Default
+		var err error
+		deletedNames, err = s.Remove(opts.names)
+		if err != nil {
+			logger.Warnf("%v", err)
```
Comment (patrickzheng200) [inline cmd/notation/key.go:235]: Should we use `logger.Errorf` here?

File: cmd/notation/sign.go [hunk 2]
```diff
@@ -100,8 +101,7 @@ func runSign(command *cobra.Command, cmdOpts *signOpts) error {
 
 func prepareSigningContent(ctx context.Context, opts *signOpts, sigRepo notationregistry.Repository) (notation.SignOptions, registry.Reference, error) {
 	ref, err := resolveReference(ctx, &opts.SecureFlagOpts, opts.reference, sigRepo, func(ref registry.Reference, manifestDesc ocispec.Descriptor) {
-		fmt.Printf("Warning: Always sign the artifact using digest(`@sha256:...`) rather than a tag(`:%s`) because tags are mutable and a tag reference can point to a different artifact than the one signed.\n", ref.Reference)
-		fmt.Printf("Resolved artifact tag `%s` to digest `%s` before signing.\n", ref.Reference, manifestDesc.Digest.String())
+		fmt.Fprintf(os.Stderr, "Warning: Always sign the artifact using digest(@sha256:...) rather than a tag(:%s) because tags are mutable and a tag reference can point to a different artifact than the one signed.\n", ref.Reference)
```
Comment (patrickzheng200) [inline cmd/notation/sign.go:104]: I'm not very sure why I could see this change in this PR. I thought we already merged PR [497](https://github.com/notaryproject/notation/pull/497)?

File: cmd/notation/verify.go [hunk 3]
```diff
@@ -72,8 +73,7 @@ func runVerify(command *cobra.Command, opts *verifyOpts) error {
 
 	// resolve the given reference and set the digest
 	ref, err := resolveReference(command.Context(), &opts.SecureFlagOpts, reference, sigRepo, func(ref registry.Reference, manifestDesc ocispec.Descriptor) {
-		fmt.Printf("Resolved artifact tag `%s` to digest `%s` before verification.\n", ref.Reference, manifestDesc.Digest.String())
-		fmt.Println("Warning: The resolved digest may not point to the same signed artifact, since tags are mutable.")
+		fmt.Fprintf(os.Stderr, "Warning: Always verify the artifact using digest(@sha256:...) rather than a tag(:%s) because resolved digest may not point to the same signed artifact, as tags are mutable.\n", ref.Reference)
```
Comment (patrickzheng200) [inline cmd/notation/verify.go:76]: Same here. Please correct me if I'm wrong, this is a change that's already been merged?

File: cmd/notation/sign.go (hunk 2 above)
Comment (priteshbandi) [reply cmd/notation/sign.go]: My branch was outdated, merged the changes now

File: cmd/notation/verify.go (hunk 3 above)
Comment (priteshbandi) [reply cmd/notation/verify.go]: Yes its jusMy branch was outdated, merged the changes now

File: cmd/notation/key.go [hunk 4]
```diff
@@ -294,26 +225,19 @@ func deleteKeys(ctx context.Context, opts *keyDeleteOpts) error {
...
-		if prevDefault == name {
-			signingKeys.Default = ""
+	var deletedNames []string 
+	var prevDefault string
+	exec := func(s *config.SigningKeys) error {
+		prevDefault = *s.Default
+		var err error
+		deletedNames, err = s.Remove(opts.names)
+		if err != nil {
+			logger.Error("something went wrong while deleting keys, error: %v", err)
```
Comment (patrickzheng200) [inline cmd/notation/key.go:235]: Maybe we could choose a more formal wording here? something like:
```suggestion
			logger.Error("Keys deletion failed to complete with error: %v", err)
```
(Btw, we'd capitalize the log messages.)

File: cmd/notation/key.go [hunk 5]
```diff
@@ -294,26 +225,18 @@ func deleteKeys(ctx context.Context, opts *keyDeleteOpts) error {
...
-		deletedNames = append(deletedNames, name)
-		if prevDefault == name {
-			signingKeys.Default = ""
+	var prevDefault string
+	exec := func(s *config.SigningKeys) error {
+		prevDefault = *s.Default
+		var err error
+		deletedNames, err = s.Remove(opts.names...)
+		if err != nil {
+			logger.Error("Keys deletion failed to complete with error: %v", err)
```
Comment (patrickzheng200) [inline cmd/notation/key.go:235]: nit:
```suggestion
			logger.Errorf("Keys deletion failed to complete with error: %v", err)
```

**Review by patrickzheng200 — APPROVED** [review]
LGTM with one nit comment

**Review by rgnote — APPROVED** [review]

**Pull request merged**